### PR TITLE
Add event delete command

### PR DESCRIPTION
**Pull request by tiongMax**
PLS DO NOT MERGE YET!

Added event delete command. The test case is not added yet.

### CODE REVIEW

**Comment by brein62** [comment]
wait for jordan's test PR then u can merge the changes, pull to your branch, and start designing the tests

File: src/test/java/seedu/address/logic/parser/IVolunteerParserTest.java [hunk 1]
```diff
@@ -50,9 +53,14 @@ public void parseCommand_clear() throws Exception {
...
+
+        Command command = parser.parseCommand(
                 VolunteerDeleteCommand.COMMAND_WORD + " " + INDEX_FIRST_VOLUNTEER.getOneBased());
-        assertEquals(new VolunteerDeleteCommand(INDEX_FIRST_VOLUNTEER), command);
+        if (command instanceof EventDeleteCommand) {
+            assertEquals(new EventDeleteCommand(INDEX_FIRST_EVENT), command);
+        } else {
+            assertEquals(new VolunteerDeleteCommand(INDEX_FIRST_VOLUNTEER), command);
+        }
     }
```
Comment (tiongMax) [inline src/test/java/seedu/address/logic/parser/IVolunteerParserTest.java:65]: This looks really dumb, but it failed without having this if else, please let me know if i did something wrong in other part.

File: src/test/java/seedu/address/logic/parser/IVolunteerParserTest.java [hunk 2]
```diff
@@ -50,9 +53,14 @@ public void parseCommand_clear() throws Exception {
...
-        VolunteerDeleteCommand command = (VolunteerDeleteCommand) parser.parseCommand(
+
+        Command command = parser.parseCommand(
                 VolunteerDeleteCommand.COMMAND_WORD + " " + INDEX_FIRST_VOLUNTEER.getOneBased());
-        assertEquals(new VolunteerDeleteCommand(INDEX_FIRST_VOLUNTEER), command);
+        if (command instanceof EventDeleteCommand) {
+            assertEquals(new EventDeleteCommand(INDEX_FIRST_EVENT), command);
+        } else {
+            assertEquals(new VolunteerDeleteCommand(INDEX_FIRST_VOLUNTEER), command);
+        }
```
Comment (brein62) [inline src/test/java/seedu/address/logic/parser/IVolunteerParserTest.java:63]: wouldnt this just run the else part? can try:

```suggestion
        VolunteerDeleteCommand command = (VolunteerDeleteCommand) parser.parseCommand(
                VolunteerDeleteCommand.COMMAND_WORD + " " + INDEX_FIRST_VOLUNTEER.getOneBased());
        assertEquals(new VolunteerDeleteCommand(INDEX_FIRST_VOLUNTEER), command);
        
        EventDeleteCommand command2 = (EventDeleteCommand) parser.parseCommand(
                EventDeleteCommand.COMMAND_WORD + " " + INDEX_FIRST_EVENT.getOneBased());
        assertEquals(new EventDeleteCommand(INDEX_FIRST_EVENT), command2);
```

tell me if there's any errors or issues with this

**Review by brein62 — CHANGES_REQUESTED** [review]
for ur consideration

File: src/test/java/seedu/address/testutil/TypicalIndexes.java [hunk 3]
```diff
@@ -9,4 +9,7 @@ public class TypicalIndexes {
     public static final Index INDEX_FIRST_VOLUNTEER = Index.fromOneBased(1);
     public static final Index INDEX_SECOND_VOLUNTEER = Index.fromOneBased(2);
     public static final Index INDEX_THIRD_VOLUNTEER = Index.fromOneBased(3);
+    public static final Index INDEX_FIRST_EVENT = Index.fromOneBased(1);
+    public static final Index INDEX_SECOND_EVENT = Index.fromOneBased(2);
+    public static final Index INDEX_THIRD_EVENT = Index.fromOneBased(3);
```
Comment (brein62) [inline src/test/java/seedu/address/testutil/TypicalIndexes.java:14]: this is surprisingly trippy

File: src/main/java/seedu/address/model/event/EventNameContainsKeywordsPredicate.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Volunteer}'s {@code Name} matches any of the keywords given.
+ */
+public class EventNameContainsKeywordsPredicate implements Predicate<Event> {
```
Comment (ruo-x) [inline src/main/java/seedu/address/model/event/EventNameContainsKeywordsPredicate.java:12]: ouh shit i did the exact same thing as u HAHAHAH telepathy?

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 5]
```diff
@@ -164,4 +165,15 @@ public static void showVolunteerAtIndex(Model model, Index targetIndex) {
...
 
+    /**
+     * Updates {@code model}'s filtered list to show only the event at the given {@code targetIndex} in the
+     * {@code model}'s address book.
+     */
+    public static void showEventAtIndex(Model model, Index targetIndex) {
+        assertTrue(targetIndex.getZeroBased() < model.getFilteredEventList().size());
+
+        Event event = model.getFilteredEventList().get(targetIndex.getZeroBased());
+        model.updateFilteredEventList(e -> e.equals(event));
```
Comment (ruo-x) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:176]: wow so smart? i did the same method but urs is faster

File: src/test/java/seedu/address/logic/parser/IVolunteerParserTest.java [hunk 6]
```diff
@@ -50,19 +51,28 @@ public void parseCommand_clear() throws Exception {
 
     @Test
     public void parseCommand_delete() throws Exception {
```
Comment (ruo-x) [inline src/test/java/seedu/address/logic/parser/IVolunteerParserTest.java:53]: perhaps u can rename this method to `parseCommand_volunteerDelete`  for clarity sake?

**Review by ruo-x — CHANGES_REQUESTED** [review]
one minor issue but well done!!

File: src/main/java/seedu/address/model/event/EventNameContainsKeywordsPredicate.java (hunk 4 above)
Comment (tiongMax) [reply src/main/java/seedu/address/model/event/EventNameContainsKeywordsPredicate.java:12]: HAHAHAH

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java (hunk 5 above)
Comment (tiongMax) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java:176]: nah Jordan helped me with this.

File: src/test/java/seedu/address/logic/parser/IVolunteerParserTest.java (hunk 6 above)
Comment (tiongMax) [reply src/test/java/seedu/address/logic/parser/IVolunteerParserTest.java:53]: gud idea

**Review by brein62 — APPROVED** [review]
LGTM!

**Review by ruo-x — APPROVED** [review]
LGTM LFG

**Pull request merged**